### PR TITLE
feat(esl-utils): Support for ESLDomElementTarget as a target for ESLResizeObserverTarget

### PR DESCRIPTION
**Pull request by fshovchko**
Target for ResizeObserverTarget with extended from 'Element' to being able to accept mixin elements.  #1562

### CODE REVIEW

File: src/modules/esl-event-listener/core/targets/resize.adapter.ts [hunk 1]
```diff
@@ -10,6 +15,14 @@ export class ESLResizeObserverTarget extends SyntheticEventTarget {
     changes.forEach(this.handleChange, this)
   );
 
+  public readonly target: Element;
+
+  private static getActualTarget(target: ESLResizeTarget): Element {
+    if (target instanceof Element) return target;
+    const host = getPropertyDescriptor(target, '$host') as Element;
```
Comment (ala-n) [inline src/modules/esl-event-listener/core/targets/resize.adapter.ts:22]: How does the property descriptor relate to the host value ? 
Is it not compatible;e with Element, and it is not something that the ESLResizeObserverTarget observer should use.

**Review by ala-n — CHANGES_REQUESTED** [review]

File: src/modules/esl-event-listener/core/targets/resize.adapter.ts [hunk 2]
```diff
@@ -1,47 +1,54 @@
...
-   * Note the {@link ESLResizeObserverTarget} instances are singletons relatively to the {@link Element}
+   * Creates {@link ESLResizeObserverTarget} for the {@link ESLResizeTarget}.
+   * Note the {@link ESLResizeObserverTarget} instances are singletons relatively to the {@link ESLResizeTarget}
    */
-  protected constructor(
-    /** Observed {@link Element} of the {@link ESLResizeObserverTarget} instance */
-    public readonly target: Element
-  ) {
+  protected constructor(target: ESLResizeTarget) {
+    if (!(target instanceof Window || target instanceof Element)) target = target.$host;
```
Comment (ala-n) [inline src/modules/esl-event-listener/core/targets/resize.adapter.ts:43]: Consider mowing to esl-utils/abstract/dom-target as utility `resolveDomTarget`
That can also be applied to other cases of ESLDomElementTarget usage

File: src/modules/esl-event-listener/core/targets/resize.adapter.ts [hunk 3]
```diff
@@ -1,47 +1,54 @@
 import {SyntheticEventTarget} from '../../../esl-utils/dom/events/target';
 import {overrideEvent} from '../../../esl-utils/dom/events/misc';
 
+import type {ESLDomElementTarget} from '../../../esl-utils/abstract/dom-target';
+
+export type ESLResizeTarget = ESLDomElementTarget | Window;
```
Comment (ala-n) [inline src/modules/esl-event-listener/core/targets/resize.adapter.ts:6]: Consider removing it due to only one actual usage, and no practical need for users as well

**Review by ala-n — CHANGES_REQUESTED** [review]
This was not an exception to creating tasks regarding ESLElementRelated and window separately.

The first one relates to creating `resolveDomTarget` basically, and it will most likely be reused 
It will be a simple change for ESLResizeObserverTarget but it relates to 
 - discussion of the `resolveDomTarget` function usage, name and etc.
 
 The second issue regarding window is a completely different story, we need to think about unified API. And it also has a chance to be even rejected.

**Review by ala-n — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Pull request merged**